examples/scales/scales.py

```diff
@@ -54,4 +57,73 @@
 ax.yaxis.set_minor_formatter(NullFormatter())
 
 
+# Function x**(1/2)
+def forward(x):
+    return x**(1/2)
+
+
+def inverse(x):
+    return x**2
+
+
+ax = axs[2, 0]
+ax.plot(x, y)
+ax.set_yscale('function', functions=(forward, inverse))
+ax.set_title('function: $x^{1/2}$')
+ax.grid(True)
+ax.yaxis.set_major_locator(FixedLocator(np.arange(0, 1, 0.2)**2))
+ax.yaxis.set_major_locator(FixedLocator(np.arange(0, 1, 0.2)))
+
+
+# Function Mercator transform
+def forward(a):
+    a = np.deg2rad(a)
+    # mask values too close to +/- 90, which are ill-defined.
+    masked = np.ma.masked_where((a <= -np.pi/2 + 0.02) |
+                                (a >= np.pi/2 - 0.02), a)
+    if masked.mask.any():
+        return np.rad2deg(
+                np.ma.log(np.abs(np.ma.tan(masked) + 1.0 / np.ma.cos(masked))))
+    else:
+        return np.rad2deg(np.log(np.abs(np.tan(a) + 1.0 / np.cos(a))))
```

Comment:
Can you just rely on this returning nan for invalid values? (possibly wrap in `with np.errstate(invalid="ignore")` to silence warnings)

Reply:
This is just an example, right?  More sophisticated users (than me) can add any error catching they want if they want to implement this...

Reply:
My point is that
```
# Function Mercator transform
def forward(a):
    a = np.deg2rad(a)
    return np.rad2deg(np.log(np.abs(np.tan(a) + 1 / np.cos(a))))
```
is much shorter, and basically works just as well (AFAICT...)

Reply:
Sure, probably.  I just stole the code from the `custom_scale.py` example to show that it was directly transferable.  But I simplified in the latest commit.